CLI/DefaultCliFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace DigipolisGent\Domainator9k\CoreBundle\CLI;
+
+use DigipolisGent\Domainator9k\CoreBundle\CLI\CliFactoryInterface;
+use DigipolisGent\Domainator9k\CoreBundle\CLI\CliInterface;
+use DigipolisGent\Domainator9k\CoreBundle\Entity\AbstractApplication;
+use DigipolisGent\Domainator9k\CoreBundle\Entity\ApplicationEnvironment;
+use DigipolisGent\Domainator9k\CoreBundle\Entity\Environment;
+use DigipolisGent\Domainator9k\CoreBundle\Entity\VirtualServer;
+use phpseclib\Crypt\RSA;
+use phpseclib\Net\SSH2;
+use Webmozart\PathUtil\Path;
+
+class DefaultCliFactory implements CliFactoryInterface
+{
+    public function create($object): ?CliInterface
+    {
+        if (!($object instanceof ApplicationEnvironment)) {
+            throw new \InvalidArgumentException(
+                static::class . ' only supports ' . ApplicationEnvironment::class
+                . ', ' . get_class($object) . ' given.'
+            );
+        }
+        $appEnv = $object;
```

Comment:
Probably because he plans on supporting other objects as well.

Reply:
Because we have to inherit from the interface, and you could theoretically create a cli for an Environment (if you have a root user and need to do something for the entire env while provisioning), VirtualServer, ... . So it can be anything, and Doctrine/Symfony entities don't have a common class they inherit from, so we can't typehint that either.